README.md

```diff
@@ -0,0 +1,400 @@
+Today `Vec<T>` implements `Deref<Target = [T]>` for convenience as this allows dispatching slice methods on objects of `Vec<T>`, ye `Deref` was meant for smart pointers instead. What this impl really feels like is more akin to unsizing. Obviously the `Deref` impl cannot be revoked from `Vec<T>` anymore, but assuming an unsizing impl would've been the proper call this would then also imply that unsizing would have to be introduced into autoderef (instead of just special casing array unsizing).
+
+What follows here is the accompanied pre-RFC:
+
+# RFC
+
+## Unsize- Feature Name: `unsizev2`
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Move the unsizing logic out of the compiler into library source, allowing for a more flexible design of the features.
+
+# Motivation
+[motivation]: #motivation
+
+Currently unsizing in Rust is very rigid, only allowed in very specific scenarios permitted by the rules surrounding the current `Unsize` and `CoerceUnsized` traits and it's automatic implementations by the compiler.
+
+This on one hand, has the downside of being very magical, as the majority of the logic happens inside the compiler opposed to the source code and it also prevents certain unsizing implementations from being doable today
+
+This RFC attempts to make these rules more flexible by also allowing user implementations of the traits that define how the metadata is derived and composed back into unsized objects.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Unsize
+
+Unsizing relationships between two types can be defined by implementing the unsafe `Unsize` trait for a type and its target unsized type.
+These implementations describe how the metadata is derived from a source type and its metadata for its target unsized type.
+
+An example implementation of `Unsize` for `[T; N]` to `[T]` unsizing looks like the following:
+
+```rs
+// SAFETY:
+// - `Unsize::target_metadata` returns length metadata that spans the entire array exactly.
+// - `[T; N]` is a contiguous slice of `T`'s, so a pointer pointing to its data is valid to be interpreted as a pointer to a slice `[T]`.
+unsafe impl<T, const N: usize> Unsize<[T]> for [T; N] {
+    fn target_metadata((): <Self as Pointee>::Metadata) -> <[T] as Pointee>::Metadata {
+        N
+    }
+}
+```
+
+The metadata for the source type `[T; N]` is the unit type `()`, as there is no metadata for sized types.
+The implementation then just returns the length `N` from the array type, as this is the appropriate metadata for a slice produced from such an array.
+
+<!-- not the best example thats following here, but I could not think of an unsized to unsized relationship thats not trait upcasting, given custom unsized typed are not currently a thing -->
```

Comment:
Maybe you can use the `Vec` example you mentioned in the motivation section?

Reply:
The example no longer fits as I could down the scope, so the vec exmaple cant be modeled with this. (but as i said, the introduction was outdated)